### DISCUSSION TITLE
Return undefined if fails to validate nativeEnum

### DISCUSSION THREAD

**Discussion by cehenriques** [colinhacks/zod#2028] [Q&A] [answered]
Hi,

using the following example:

```ts
enum Fruits {
  Apple = "apple",
  Banana = "banana"
}

const schema = z
      .object({
        fruit: z.nativeEnum(Fruits),
      })
      .partial();

const parsedSchema = schema.safeParse("");
```

How can I have zod return undefined if it fails to parse the nativeEnum, like in the example by just providing an empty string?

Thank you.

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
enum Fruits {
    Apple = 'apple',
    Banana = 'banana'
}

const schema = z.object( {
    fruit: z.nativeEnum( Fruits ),
} ).partial().optional().catch( undefined )

console.log( schema.parse( '' ) ) // undefined
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by cehenriques** [reply]
I was trying `z.nativeEnum( Fruits ).catch(undefined)` but I was getting an error, but this way it works.
Thanks for the help.
